lib/tasks/valid_test_data_generator.rb

```diff
@@ -108,7 +108,37 @@ def weighted_choice(selection:, odds:)
   class AmbitionSpecificPopulater < LeadProviderPopulater
     class << self
       def call(name:, total_schools: 3, participants_per_school: 3000)
-        new(name: name).call(total_schools: total_schools, participants_per_school: participants_per_school)
+        generator = new(name: name)
+        generator.remove_old_data(created_at: "2021-08-18 13:43".."2021-08-18 13:49")
+        generator.call(total_schools: total_schools, participants_per_school: participants_per_school)
+      end
+    end
+
+    def remove_old_data(created_at:)
+      participants = lead_provider.ecf_participants.where(created_at: created_at)
+      participants.all.each do |participant|
+        tp = participant.teacher_profile
+        # Just in case the withdrawn merge happens before this is run.
+        # tp.participant_profiles.each do |pp|
+        #   pp.participant_profile_states.destroy_all!
+        # end
+        tp.destroy!
+        participant.destroy!
+      end
+      schools = lead_provider.schools.where(created_at: created_at)
+      schools.all.each do |school|
+        partnership = Partnership.find_by(lead_provider: lead_provider, school: school, cohort: Cohort.current)
+        delivery_partner = partnership.delivery_partner
+        provider_relationship = ProviderRelationship.find_by(lead_provider: lead_provider,
+                                                             cohort: Cohort.current,
+                                                             delivery_partner: delivery_partner)
+        provider_relationship.destroy!
+        partnership.destroy!
+        delivery_partner.destroy!
+        school_cohort = SchoolCohort.find_by(school: school, cohort: Cohort.current, induction_programme_choice: "full_induction_programme")
+        school_cohort.ecf_participant_profiles.destroy_all
```

Comment:
Likewise here, I think `has_many :participant_profiles, dependent: :destroy` in teacher profile will take care of them before you reach this line.

Reply:
Same comment. I don't mind extra code when it's something that I may forget later.